### PR TITLE
Don't relayout a Text if only its color changed.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Review by jason-simmons — APPROVED** [review]

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -64,11 +64,20 @@ class RenderParagraph extends RenderBox {
...
     assert(value != null);
-    if (_textPainter.text == value)
-      return;
-    _textPainter.text = value;
-    _overflowShader = null;
-    markNeedsLayout();
+    switch (_textPainter.text.compareTo(value)) {
+      case RenderComparison.identical:
+      case RenderComparison.metadata:
+        return;
```
Comment (HansMuller) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Why a return here but a break below?

**Comment by HansMuller** [comment]
LGTM. Nice use of enum.index.

File: packages/flutter/lib/src/rendering/paragraph.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/rendering/paragraph.dart]: accident of how i wrote the code, though i can justify it post-hoc: the "return" is an early return (nothing changed, bail out), whereas the other two branches are actually doing something. If we ever add something after the switch, I'd expect the identical/metadata cases to do nothing, but the other cases to do it.

File: packages/flutter/lib/src/painting/basic_types.dart [hunk 2]
```diff
@@ -52,3 +52,41 @@ export 'dart:ui' show
...
+/// it will affect the rendering.
+///
+/// Used by [TextSpan.compareTo] and [TextStyle.compareTo].
+///
+/// The values in this enum are ordered such that they are in increasing order
+/// of cost. A value with index N implies all the values with index less than N.
+/// For example, [layout] (index 3) implies [paint] (2).
+enum RenderComparison {
+  /// The two objects are identical (meaning deeply equal, not necessarily
+  /// [identical]).
```
Comment (tvolkert) [inline packages/flutter/lib/src/painting/basic_types.dart]: This just hyperlinks right back to this enum value (as seen on https://master-docs-flutter-io.firebaseapp.com/flutter/painting/RenderComparison-class.html)

Comment (Hixie) [inline packages/flutter/lib/src/painting/basic_types.dart]: oops. I guess I want [dart:io.identical]. (cc @jcollins-g)

Comment (Hixie) [inline packages/flutter/lib/src/painting/basic_types.dart]: Hopefully fixed in https://github.com/flutter/flutter/pull/11330

**Pull request merged**